frontend/app-development/features/appPublish/components/Deploy/DeployDropdown/DeployDropdown.test.tsx

```diff
@@ -0,0 +1,207 @@
+    label: textMock('app_deployment.version_label', {
+      tagName: appReleases[1].tagName,
+      createdDateTime: created,
+    }),
+    value: appReleases[1].tagName,
+  },
+];
+
+describe('DeployDropdown', () => {
+  afterEach(jest.clearAllMocks);
+
+  it('renders a spinner while loading data', () => {
+    render();
+
+    expect(screen.getByTitle(textMock('app_deployment.releases_loading'))).toBeInTheDocument();
+  });
+
+  it('renders an error message if an error occurs while loading data', async () => {
+    render(
+      {},
+      {
+        getAppReleases: jest.fn().mockImplementation(() => Promise.reject()),
+      },
+    );
+    await waitForElementToBeRemoved(() =>
+      screen.queryByTitle(textMock('app_deployment.releases_loading')),
+    );
+
+    expect(screen.getByText(textMock('app_deployment.releases_error'))).toBeInTheDocument();
+  });
+
+  it('render no image options message when image options are empty', async () => {
+    const user = userEvent.setup();
+
+    render(
+      {},
+      {
+        getAppReleases: jest.fn().mockImplementation(() =>
+          Promise.resolve({
+            results: [],
+          }),
+        ),
+      },
+    );
+    await waitForElementToBeRemoved(() =>
+      screen.queryByTitle(textMock('app_deployment.releases_loading')),
+    );
+
+    const select = await screen.findByLabelText(textMock('app_deployment.choose_version'));
+    await user.click(select);
+
+    expect(screen.getByText(textMock('app_deployment.no_versions'))).toBeInTheDocument();
+  });
+
+  it('renders image options', async () => {
+    const user = userEvent.setup();
+
+    render();
+    await waitForElementToBeRemoved(() =>
+      screen.queryByTitle(textMock('app_deployment.releases_loading')),
+    );
+
+    const select = screen.getByLabelText(textMock('app_deployment.choose_version'));
+    await user.click(select);
+
+    expect(screen.getByRole('option', { name: imageOptions[0].label })).toBeInTheDocument();
+    expect(screen.getByRole('option', { name: imageOptions[1].label })).toBeInTheDocument();
+  });
+
+  it('selects default image option', async () => {
+    render({ selectedImageTag: imageOptions[0].value });
+    await waitForElementToBeRemoved(() =>
+      screen.queryByTitle(textMock('app_deployment.releases_loading')),
+    );
+
+    expect(screen.getByRole('combobox')).toHaveValue(imageOptions[0].label);
+  });
+
+  it('selects new image option', async () => {
+    const user = userEvent.setup();
+
+    render();
+    await waitForElementToBeRemoved(() =>
+      screen.queryByTitle(textMock('app_deployment.releases_loading')),
+    );
+
+    const select = screen.getByLabelText(textMock('app_deployment.choose_version'));
+    await user.click(select);
+
+    const option = screen.getByRole('option', { name: imageOptions[1].label });
+    await user.click(option);
+
+    await waitFor(() => {
+      expect(defaultProps.setSelectedImageTag).toHaveBeenCalledWith(imageOptions[1].value);
+    });
```

Comment:
Do we need `await waitFor` here? 🤔

Reply:
It seems so. Tried to remove it, and the tests fails then. 
The code is just copied from another component when I refactored, so have not seem to much into it. I can try to refactor the test a bit, but I believe that piece of code is needed 🤔